src/components/waveControl.tsx

```diff
@@ -1,28 +1,26 @@
-import React from 'react';
-import styled from "styled-components";
-import waveforms from "../data/waveforms";
+import React, { useContext } from 'react';
+import { getAllWaveTypes } from "../data/waveforms";
+
+import { KeyboardContext, UpdateKeyboardContext } from "../hooks/keyboardContext";
 
-interface WaveControlProps {
-    setWave: (wave: string) => void
-}
 
-function onWaveSelectClick(e: any, setWaveFunc: (wave: string) => void) {
-    var selectedWave = e.target.value;
-    if (selectedWave != null) {
-        setWaveFunc(selectedWave);
+function WaveControl(){
+    const setState = useContext(UpdateKeyboardContext);
+    const state = useContext(KeyboardContext);
```

Comment:
Is it worth collapsing these two lines into some helper function like `{ state, setState } = getKeyboardContext()`?

Reply:
Hmm - maybe? And if we do that I'd want it in a separate file so it can be used by all components that both get and set. With our current use of setState(), we need to get every time set set. 

That said, I don't know if we can extract it, since useContext is render specific. It only works here since the component is being rendered to the dom.

Reply:
Sure, I think I see what you're saying

Reply:
I say we leave it as is for now, but I have thoughts